src/ess/dream/io/geant4.py

```diff
@@ -148,10 +148,10 @@ def _split_detectors(
 def _extract_detector(
     detector_groups: sc.DataArray, detector_id_name: str, detector_id: sc.Variable
 ) -> Optional[sc.DataArray]:
-    try:
-        return detector_groups[detector_id_name, detector_id].value
-    except IndexError:
+    events = detector_groups[detector_id_name, detector_id].value
+    if len(events) == 0:
         return None
```

Comment:
Won't downstream code break if some detectors are missing/none?

Reply:
Isn't this is how the nexus loader behaves?
Stuff may break but it seems cleaner to not include a detector in the group if the file also didn't include it.